### PR TITLE
Use the updated Destination

### PR DESCRIPTION
**Pull request by Harwayne**
Related to knative/eventing#1918

- Update PullSubscription to use SourceSpec. 
- Update the following to use the new Destination from knative.dev/pkg.
    - PubSub
    - PullSubscription
    - Scheduler
    - Storage

### CODE REVIEW

**Comment by Harwayne** [comment]
Ready for review, PTAL.

File: pkg/apis/pubsub/v1alpha1/pull_subscription_types.go [hunk 1]
```diff
@@ -59,6 +59,9 @@ var _ = duck.VerifyType(&PullSubscription{}, &duckv1.Conditions{})
 
 // PullSubscriptionSpec defines the desired state of the PullSubscription.
 type PullSubscriptionSpec struct {
+	// This brings in CloudEventOverrides and Sink.
+	duckv1.SourceSpec
```
Comment (n3wscott) [inline pkg/apis/pubsub/v1alpha1/pull_subscription_types.go]: I think this needs a json inline tag here

File: config/300-pubsub.yaml [hunk 2]
```diff
@@ -65,6 +65,41 @@ spec:
...
+                        apiVersion:
+                          type: string
+                          minLength: 1
+                        kind:
+                          type: string
+                          minLength: 1
+                        name:
+                          type: string
+                          minLength: 1
+                - type: object
```
Comment (grantr) [inline config/300-pubsub.yaml]: Should this one also have the `required` from above?

File: config/300-pubsub.yaml [hunk 3]
```diff
@@ -65,6 +65,41 @@ spec:
...
               type: object
               description: "Reference to an object that will resolve to a domain name to use as the sink."
+              anyOf:
+                - type: object
+                  properties:
+                    uri:
+                      type: string
+                      minLength: 1
+                - type: object
+                  properties:
```
Comment (grantr) [inline config/300-pubsub.yaml]: Should this list also include a uri field?

File: pkg/apis/pubsub/v1alpha1/pull_subscription_types.go (hunk 1 above)
Comment (Harwayne) [reply pkg/apis/pubsub/v1alpha1/pull_subscription_types.go]: Done.

File: config/300-pubsub.yaml (hunk 3 above)
Comment (Harwayne) [reply config/300-pubsub.yaml]: Isn't that the first 'branch' of the anyOf?

File: config/300-pubsub.yaml (hunk 2 above)
Comment (Harwayne) [reply config/300-pubsub.yaml]: Probably, but this is the deprecated version. Would you like me to do so?

**Comment by Harwayne** [comment]
/hold cancel

Fixed the issue (at least I think so...).

File: config/300-pubsub.yaml (hunk 2 above)
Comment (grantr) [reply config/300-pubsub.yaml]: I won't complain either way :smiley:

**Pull request merged**